### ISSUE TITLE
Port validator_service.py to web3.py

### ISSUE THREAD

**Issue by djrtwo** [ethereum/casper#48]
Due to the pending deprecation of the pethereum/pyethapp stack, I propose porting existing `validator_service.py` functionality to use web3.py. Not only is web3.py well-featured and actively developed, web3.py can plug into any client that conforms to the standard json-rpc interface. This is particularly exciting because a web3.py validator_service will be able to plug into any of the major clients (yay code reuse).

# Proposed Implementation
I haven't fully spec'd this out yet, but it should be a natural translation from pyethapp to web3.py. More details to follow.

- port functionality 1:1 from pyethapp to web3.py
- new service will live in this repo
- move contract tests to `tests/contracts`
- add new service tests at `tests/service`

**Comment by ChihChengLiang** [comment]
Note that other clients still need to implement features such as:
- Casper Fork Choice Rule
- Epoch initialization
- Casper Vote Gas Refunds
- Casper Block Ordering

specified in the [spec](http://notes.ethereum.org/GwTgLAZghgRgHGAtABmARgKaLFNzEIwjZgDsAxsBAMzIwbJpA===?both=#outline)

**Comment by djrtwo** [comment]
Absolutely. In the issue, I should note that this is for when other clients implement the casper spec.

**Comment by djrtwo** [comment]
`decimal10` support in web3.py is currently a blocker
https://github.com/ethereum/web3.py/issues/700

**Comment by pipermerriam** [comment]
Someone can start on `decimal10` abi support now.  The implementation should be done in [`eth-abi`](https://github.com/ethereum/eth-abi) and should be reasonably straight forward to do if you dig in and look at how the current encoders/decoders are written.

**Comment by djrtwo** [comment]
`eth-abi` library is waiting on decision to be made on whether vyper is going to keep decimal10 or just move to support fixedNxM

https://github.com/ethereum/vyper/issues/566
